numpy/core/src/multiarray/ctors.c

```diff
@@ -1259,6 +1259,52 @@ PyArray_New(PyTypeObject *subtype, int nd, npy_intp *dims, int type_num,
     return new;
 }
 
+int
+_copy_from_pil_style(char* dest, int ndim, char *buf, npy_intp *shape,
+                     npy_intp *strides, npy_intp *suboffsets, npy_intp itemsize)
+{
+    int ax, last_axis;
+    npy_intp indices[NPY_MAXDIMS];
+    void **pointers[NPY_MAXDIMS];
+    char *pointer = (char*)buf;
+    npy_intp blocksize;
+    /* Initialize indices and pointers. */
+    for (ax = 0; ax < ndim; ++ax) {
+        indices[ax] = 0;
+        pointers[ax] = (void **)pointer;
+        if (suboffsets[ax] >= 0) {
+            last_axis = ax;
+	        pointer = *((char**)pointer) + suboffsets[ax];
+	    }
+    }
+    /* compute blocksize */
+    blocksize = itemsize;
+    for (ax = last_axis + 1; ax < ndim; ++ax)
```

Comment:
If all `suboffsets[ax] < 0` then `last_axis` never gets initialized when we get here. Even if that is not really possible (is it not?) the compiler does not seem able to figure it out, and that's what's making Travis unhappy. I guess the right value to initialize it at the beginning of the function is `-1`?